spec.html

```diff
@@ -13524,70 +13526,58 @@ <h1>
+        1. If _hasParameterExpressions_ is *true*, then
           1. Let _varEnv_ be NewDeclarativeEnvironment(_env_).
           1. Set the VariableEnvironment of _calleeContext_ to _varEnv_.
           1. Let _instantiatedVarNames_ be a new empty List.
-          1. For each element _n_ of _varNames_, do
-            1. If _n_ is not an element of _instantiatedVarNames_, then
-              1. Append _n_ to _instantiatedVarNames_.
-              1. Perform ! _varEnv_.CreateMutableBinding(_n_, *false*).
-              1. If _n_ is not an element of _parameterBindings_ or if _n_ is an element of _functionNames_, let _initialValue_ be *undefined*.
-              1. Else,
-                1. Let _initialValue_ be ! _env_.GetBindingValue(_n_, *false*).
-              1. Perform ! _varEnv_.InitializeBinding(_n_, _initialValue_).
-              1. NOTE: A var with the same name as a formal parameter initially has the same value as the corresponding initialized parameter.
+        1. Else,
+          1. Let _varEnv_ be _env_.
+          1. Let _instantiatedVarNames_ be a copy of the List _parameterBindings_.
+        1. For each element _n_ of _varNames_, do
+          1. If _n_ is not an element of _instantiatedVarNames_, then
+            1. Append _n_ to _instantiatedVarNames_.
+            1. Perform ! _varEnv_.CreateMutableBinding(_n_, *false*).
+            1. NOTE: A var with the same name as a formal parameter initially has the same value as the corresponding initialized parameter.
+            1. If _n_ is an element of _parameterBindings_ and _n_ is not an element of _functionNames_, let _initialValue_ be ! _env_.GetBindingValue(_n_, *false*). Otherwise, let _initialValue_ be *undefined*.
+            1. Perform ! _varEnv_.InitializeBinding(_n_, _initialValue_).
         1. [id="step-functiondeclarationinstantiation-web-compat-insertion-point"] NOTE: Annex <emu-xref href="#sec-web-compat-functiondeclarationinstantiation"></emu-xref> adds additional steps at this point.
-        1. If _strict_ is *false*, then
-          1. Let _lexEnv_ be NewDeclarativeEnvironment(_varEnv_).
-          1. NOTE: Non-strict functions use a separate Environment Record for top-level lexical declarations so that a direct eval can determine whether any var scoped declarations introduced by the eval code conflict with pre-existing top-level lexically scoped declarations. This is not needed for strict functions because a strict direct eval always places all declarations into a new Environment Record.
-        1. Else, let _lexEnv_ be _varEnv_.
+        1. NOTE: Non-strict functions use a separate Environment Record for top-level lexical declarations so that a direct eval can determine whether any var scoped declarations introduced by the eval code conflict with pre-existing top-level lexically scoped declarations. This is not needed for strict functions, in which direct eval declarations never affect the outer environment.
```

Comment:
"direct eval declarations never affect the outer environment" isn't true - it can't create new bindings, but it can alter the values of existing ones. The old wording seems fine to me.

(Sidebar: it's quite hard to notice when you move code and simultaneously tweak it. For reviewability purposes I'd appreciate it if in the future you could call such things out, or split them into distinct commits.)

Reply:
```suggestion
        1. NOTE: Non-strict functions use a separate Environment Record for top-level lexical declarations so that a direct eval can determine whether any var scoped declarations introduced by the eval code conflict with pre-existing top-level lexically scoped declarations. This is not needed for strict functions, in which direct eval declarations never add bindings to the outer environment.
```